### PR TITLE
[BUGFIX] Fallback to query when metric labels are empty

### PR DESCRIPTION
**Pull request by sjcobb**
Fallback to query name when metric labels are empty

### Before
image

### After
image

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).
- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.

### CODE REVIEW

File: ui/core/src/utils/is-empty-object.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Determines if an object is an empty object
+ */
+export function isEmptyObject(obj: object) {
```
Comment (saminzadeh) [inline ui/core/src/utils/is-empty-object.ts:17]: Even though this is small, my vote would be not to rewrite utils and start using lodash here. 
https://lodash.com/docs/4.17.15#isEmpty

We can treeshake the library to only use utils that are consumed in the code.

**Review by saminzadeh — APPROVED** [review]

**Comment by saminzadeh** [comment]
The tooltip has a strange name?
image

Shouldn't it be the full query name?

**Comment by sjcobb** [comment]
> The tooltip has a strange name? image
> 
> Shouldn't it be the full query name?

Yeah it should, will be a quick fix, thanks for catching that!

File: ui/components/src/TimeSeriesTooltip/SeriesInfo.tsx [hunk 2]
```diff
@@ -23,10 +23,11 @@ export interface SeriesInfoProps {
   markerColor: string;
   totalSeries: number;
   wrapLabels?: boolean;
+  showQuery?: boolean;
 }
 
 export function SeriesInfo(props: SeriesInfoProps) {
-  const { seriesName, formattedY, markerColor, totalSeries, wrapLabels } = props;
+  const { seriesName, formattedY, markerColor, totalSeries, wrapLabels, showQuery = true } = props;
```
Comment (sjcobb) [inline ui/components/src/TimeSeriesTooltip/SeriesInfo.tsx:30]: Added this `showQuery` prop for additional customization of series name display in the tooltip, but I need to expose this on LineChart itself. 

Ideal behavior in the future will probably be to only show the query automatically when more than 1 query was used in the panel.

File: ui/core/src/utils/is-empty-object.ts (hunk 1 above)
Comment (sjcobb) [reply ui/core/src/utils/is-empty-object.ts:17]: Yeah this makes sense, our lodash usage is a mess though, I'm planning to fix this up when I tackle Issue #894 (see error in comment [here](https://github.com/perses/perses/issues/894#issuecomment-1363612315))

Lodash also isn't a dependency in prometheus-plugin yet, so I want to make sure we set it up properly this time

Comment (sjcobb) [reply ui/core/src/utils/is-empty-object.ts:17]: cc @juliepagano we might want to add this to our list of technical debt / deciding long term what we want to do about lodash usage, I'll add it as a discussion topic too

File: ui/components/src/TimeSeriesTooltip/SeriesInfo.tsx (hunk 2 above)
Comment (sjcobb) [reply ui/components/src/TimeSeriesTooltip/SeriesInfo.tsx]: Fixed this up and added a new optional tooltipConfig prop to customize other TimeSeriesTooltip behavior

**Pull request merged**